app/services/puppet_fact_parser.rb

```diff
@@ -27,14 +27,14 @@ def operatingsystem
       elsif os_name[/FreeBSD/i]
         orel.gsub!(/\-RELEASE\-p[0-9]+/, '')
       end
-      major, minor = orel.split(".")
-      major.to_s.gsub!(/\D/, '') unless is_numeric? major
-      minor.to_s.gsub!(/\D/, '') unless is_numeric? minor
-      args = {:name => os_name, :major => major.to_s, :minor => minor.to_s}
-      os = Operatingsystem.where(args).first || Operatingsystem.create!(args)
```

Comment:
No reason to save the model twice for new OSes, if there is a failing validation it will be thrown in the `.save!` below.